zee-grammar/src/config.rs

```diff
@@ -17,6 +17,8 @@ pub struct ModeConfig {
     pub comment: Option<CommentConfig>,
     pub indentation: IndentationConfig,
     pub grammar: Option<GrammarConfig>,
+    #[serde(default)]
+    pub shebangs: Vec<String>,
```

Comment:
I was thinking whether this should be a new field, rather than a new variant of `FilenamePattern` -- granted, we should probably rename it to `FilePattern` as it will look inside the file too to determine if the mode applies. I.e. something like `FilePattern::Shebang`. 

The structure I suggest may make it harder to avoid reading the file when the filename would suffice.